libc/src/__support/CPP/atomic.h

```diff
@@ -166,7 +172,7 @@ LIBC_INLINE void atomic_thread_fence([[maybe_unused]] MemoryOrder mem_ord) {
 // except no instructions for memory ordering are issued. Only reordering of
 // the instructions by the compiler is suppressed as order instructs.
 LIBC_INLINE void atomic_signal_fence([[maybe_unused]] MemoryOrder mem_ord) {
-#if LIBC_HAS_BUILTIN(__atomic_signal_fence)
+#if __has_builtin(__atomic_signal_fence)
```

Comment:
So, I think `if constexpr` has the issue that it doesn't necessarily cull the AST if the related builtin wasn't found. So you can get cases where it will still error or emit a warning.

Reply:
AFAICT it does
https://godbolt.org/z/ojcfTTcdM

Reply:
Ok my bad, your godbolt link is the one I thought I sent (bad night, need more coffee). You're right that all  statements must be well formed even if discarded. My point is moot.